sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosOperations.java

```diff
@@ -123,6 +125,28 @@ public interface CosmosOperations {
      */
     <T> T insert(T objectToSave, PartitionKey partitionKey);
 
+    /**
+     * patches item
+     * @param id must not be {@literal null}
+     * @param partitionKey must not be {@literal null}
+     * @param domainType must not be {@literal null}
+     * @param patchOperations must not be {@literal null}, max operations is 10
+     * @param <T> type class of domain type
+     * @return the patched item
+     */
+    <T> T patch(Object id, PartitionKey partitionKey, Class<T> domainType, CosmosPatchOperations patchOperations);
+
+    /**
+     * patches item
+     * @param id must not be {@literal null}
+     * @param partitionKey must not be {@literal null}
+     * @param domainType must not be {@literal null}
+     * @param patchOperations must not be {@literal null}, max operations is 10
+     * @param options Optional CosmosPatchItemRequestOptions, e.g. options.setFilterPredicate("FROM products p WHERE p.used = false");
+     * @param <T> type class of domain type
+     * @return the patched item
+     */
+    <T> T patch(Object id, PartitionKey partitionKey, Class<T> domainType, CosmosPatchOperations patchOperations, CosmosPatchItemRequestOptions options);
```

Comment:
If this has been discussed in the API review please just add brief comment and resolve this. But why are we adding an overload with request options here when none of the other operations allow specifying request options? I am not against it - just curious why we now decide t add it and skipped it for all other operations? Is there a more common scenario that would need it? Or is the idea to eventually add overload to other operations with request options as well?

Reply:
But most of the discussion was around whether to add an overload, or force called to pass null (we went with overload, so that the method call in most cases will be less verbose). But I think here you are asking why we add request options at all since we don't expose that elsewhere, which is a good question? It is purely to facilitate setFilterPredicate for patch. If we generally don't want to expose request options, then I guess we should encapsulate this and just have a string parameter in the overload, something like String filterPredicate.

Reply:
@lmolkova - actually that's what we had earlier :)
We had one API with `requestOptions` included and `requestOptions` can be nullable / optional. However, we got initial customer feedback from the PR review that if they need to pass in null, why are we forcing them to pass null and we should expose one API for simple use case. That's why we switched back to having 2 APIs.

Reply:
two thanks for the clarification, @kushagraThapar ! Two methods are fine (one with options, one without). I meant that the verbose one (with extra options), should take an extendable options class and not an individual `filterPredicate` parameter.

Reply:
I think that is just in the example / doc on the API, not an actual API itself, no?